app/controllers/candidate_interface/references/review_controller.rb

```diff
@@ -62,7 +62,14 @@ def destroy
           redirect_to_review_page and return
         end
 
-        @reference.destroy!
+        if FeatureFlag.active?(:reference_selection)
+          DeleteReference.new.call(reference: @reference)
+          if @reference.selected
+            current_application.update!(references_completed: false)
+          end
+        else
+          @reference.destroy!
+        end
```

Comment:
We could move this `if` block into `DeleteReference`, so the controller only has to make the single-line call.